src/main/java/demurebot/DemureBot.java

```diff
@@ -0,0 +1,96 @@
+package demurebot;
+
+import demurebot.command.Command;
+import demurebot.storage.Storage;
+import demurebot.task.TaskList;
+import demurebot.ui.Parser;
+import demurebot.ui.Ui;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Main class for the DemureBot application.
+ * This class handles the initialization and execution of the chatbot.
+ */
+public class DemureBot {
+    private TaskList list;
+    private final Storage storage;
+    private final Ui ui;
+
+    /**
+     * Constructs a new DemureBot instance.
+     * Initializes the storage and UI components.
+     */
+    public DemureBot() {
+        this.storage = new Storage();
+        this.ui = new Ui();
+        this.list = null;
+    }
+
+    /**
+     * Runs the main loop of the DemureBot application.
+     * Handles user input, command execution, and data persistence.
+     */
+    private void run() {
+        Scanner scanner = new Scanner(System.in);
+
+        // check if user ended session
+        boolean isFinished = false;
+
+        // introduction to chatbot
+        this.ui.displayStart();
+
+        // check if data folder exists if not create it
+        String folderPath = "./data";
+        this.storage.checkFolder(folderPath);
+
+        // check if saved data exists if not create it
+        String filePath = "./data/tasks.txt";
+        boolean savedDataExists = this.storage.checkFile(filePath);
+
+        // load saved data if exists
+        if (savedDataExists) {
```

Comment:
Perhaps a better naming convention for this variable is to start with 'is' or 'has' to indicate that it is a boolean variable. a suggestion "hasSavedData"